app/scripts/lib/rpc-method-middleware/createMethodMiddleware.js

```diff
@@ -1,19 +1,31 @@
-import { permissionRpcMethods } from '@metamask/permission-controller';
 import { rpcErrors } from '@metamask/rpc-errors';
 import { selectHooks } from '@metamask/snaps-rpc-methods';
 import { hasProperty } from '@metamask/utils';
-import { handlers as localHandlers, legacyHandlers } from './handlers';
 
-const allHandlers = [...localHandlers, ...permissionRpcMethods.handlers];
+import {
+  handlers as localHandlers,
+  eip1193OnlyHandlers,
+  ethAccountsHandler,
+} from './handlers';
+import { getPermissionsHandler } from './handlers/wallet-getPermissions';
+import { requestPermissionsHandler } from './handlers/wallet-requestPermissions';
+import { revokePermissionsHandler } from './handlers/wallet-revokePermissions';
 
-// The primary home of RPC method implementations in MetaMask. MUST be subsequent
-// to our permissioning logic in the JSON-RPC middleware pipeline.
-export const createMethodMiddleware = makeMethodMiddlewareMaker(allHandlers);
+// The primary home of RPC method implementations for the injected 1193 provider API. MUST be subsequent
+// to our permissioning logic in the EIP-1193 JSON-RPC middleware pipeline.
+export const createEip1193MethodMiddleware = makeMethodMiddlewareMaker([
+  ...localHandlers,
+  ...eip1193OnlyHandlers,
+  getPermissionsHandler,
```

Comment:
Nit: Maybe we can add a comment to explain what this group is:
```suggestion
  // EIP-2255 Permission handlers
  getPermissionsHandler,
```